activerecord/test/cases/batches_test.rb

```diff
@@ -366,16 +442,40 @@ def test_in_batches_should_be_loaded
     end
   end
 
+  def test_in_batches_should_not_be_loaded_when_primary_key_is_title
+    Zine.primary_key = 'title'
+
+    Zine.in_batches(of: 1) do |relation|
+      assert_not relation.loaded?
+    end
+
+    Zine.in_batches(of: 1, load: false) do |relation|
+      assert_not relation.loaded?
+    end
+  ensure
+    Zine.primary_key = 'id'
+  end
+
+  def test_in_batches_should_be_loaded_when_primary_key_is_title
```

Comment:
It is also a order dependent problem. I think you need to all `reset_column_information`.